docs/quickstarts/react-router.mdx

```diff
@@ -0,0 +1,270 @@
+---
+title: React Router Quickstart
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your React Router application.
+---
+
+<TutorialHero
+  framework="react-router"
+  exampleRepo={[
+    {
+      title: "React Router Quickstart Repo",
+      link: "https://github.com/clerk/clerk-react-router-quickstart"
+
+    }
+  ]}
+  beforeYouStart={[
+    {
+      title: "Set up a Clerk application",
+      link: "/docs/quickstarts/setup-clerk",
+      icon: "clerk",
+    },
+    {
+      title: "Install React Router as a framework",
+      link: "https://reactrouter.com/start/framework/installation",
+      icon: "react-router",
+    }
+  ]}
+>
+  - Install `@clerk/react-router`
+  - Set your Clerk API keys
+  - Configure `rootAuthLoader`
+  - Add `<ClerkProvider>`
+  - Add Clerk components
+</TutorialHero>
+
+Learn how to use Clerk to quickly and easily add secure authentication and user management to your React Router app. This guide assumes that you are using React Router v7 or later.
+
+<Steps>
+  ### Install `@clerk/react-router`
+
+  Clerk's React Router SDK gives you access to prebuilt components, hooks, and helpers to make user authentication easier.
+
+  Run the following command to install the SDK:
+
+  <CodeBlockTabs options={["npm", "yarn", "pnpm"]}>
+    ```bash {{ filename: 'terminal' }}
+    npm install @clerk/react-router
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    yarn add @clerk/react-router
+    ```
+
+    ```bash {{ filename: 'terminal' }}
+    pnpm add @clerk/react-router
+    ```
+  </CodeBlockTabs>
+
+  ### Set your Clerk API keys
+
+  <SignedIn>
+    Add the following keys to your `.env` file. These keys can always be retrieved from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page of your Clerk Dashboard.
+  </SignedIn>
+
+  <SignedOut>
+    1. Navigate to the Clerk Dashboard.
+    1. In the navigation sidebar, select [API Keys](https://dashboard.clerk.com/last-active?path=api-keys).
+    1. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+    1. Paste your keys into your `.env` file.
+
+    The final result should resemble the following:
+  </SignedOut>
+
+  ```env {{ filename: '.env' }}
+  CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  ### Configure `rootAuthLoader`
+
+  To configure Clerk in your React Router app, you will need to update your root loader. This will enable you to have access to authentication state in any React Router route.
+
+  Update your `root.tsx` file with the following code:
```

Comment:
Ref docs: https://reactrouter.com/explanation/special-files#roottsx

The file contents below are the output after following their https://reactrouter.com/start/framework/installation step